lib/lrama/command.rb

```diff
@@ -5,65 +5,104 @@ class Command
     LRAMA_LIB = File.realpath(File.join(File.dirname(__FILE__)))
     STDLIB_FILE_PATH = File.join(LRAMA_LIB, 'grammar', 'stdlib.y')
 
-    def run(argv)
-      parser_options(argv)
+    def initialize(argv)
+      @logger = Lrama::Logger.new
+      @options = OptionParser.parse(argv)
+      @tracer = Tracer.new(@logger, **@options.trace_opts)
+      @reporter = Reporter.new(**@options.report_opts)
```

Comment:
I prefer to pass `io` when `Reporter` is initialized.

Reply:
If I pass the result of File.open here and initialize it, it will be difficult to manage when I let it abort in the middle of the process, so I'm doing it the way it is now, but would it be better to initialize it here?

Reply:
Understand your point.